<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [antlr-interest] I want to throw an exception and stop parse,	please!
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:antlr-interest%40antlr.org?Subject=Re:%20%5Bantlr-interest%5D%20I%20want%20to%20throw%20an%20exception%20and%20stop%20parse%2C%0A%09please%21&In-Reply-To=%3CF5CB4909-1BC3-4D2B-A981-4178990BEDEF%40stud.hs-heilbronn.de%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="034643.html">
   <LINK REL="Next"  HREF="034645.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[antlr-interest] I want to throw an exception and stop parse,	please!</H1>
    <B>Tilman Bender</B> 
    <A HREF="mailto:antlr-interest%40antlr.org?Subject=Re:%20%5Bantlr-interest%5D%20I%20want%20to%20throw%20an%20exception%20and%20stop%20parse%2C%0A%09please%21&In-Reply-To=%3CF5CB4909-1BC3-4D2B-A981-4178990BEDEF%40stud.hs-heilbronn.de%3E"
       TITLE="[antlr-interest] I want to throw an exception and stop parse,	please!">tbender at stud.hs-heilbronn.de
       </A><BR>
    <I>Sun May 31 03:21:53 PDT 2009</I>
    <P><UL>
        <LI>Previous message: <A HREF="034643.html">[antlr-interest] I want to throw an exception and stop parse, please!
</A></li>
        <LI>Next message: <A HREF="034645.html">[antlr-interest] I want to throw an exception and stop parse,	please!
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#34644">[ date ]</a>
              <a href="thread.html#34644">[ thread ]</a>
              <a href="subject.html#34644">[ subject ]</a>
              <a href="author.html#34644">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Hi Kevin,

Yes you can, but it is a bit of a dirty hack:
Make your custom Exception subclass RuntimeException.
This has two effects:

- The compiler will allow you to throw it in the recover method  
allthough recover has no throws declaration (RuntimeExceptions are  
unchecked)
- You cannot directly catch MyException in the code that invoked your  
parser, since the compiler will complain, that none such Exception is  
declared to
be thrown.

So that means:  Subclass RuntimeException and in the code calling your  
parser catch RuntimeException and gracefully exit your compiler/ 
interpreter/whatever.

HTH

Tilman Bender
Student des Software Engineering
Hochschule Heilbronn
<A HREF="http://www.antlr.org/mailman/listinfo/antlr-interest">tbender at stud.hs-heilbronn.de</A>



Am 31.05.2009 um 08:21 schrieb Kevin Chen:

&gt;<i> But, there's no throws in original recover method:
</I>&gt;<i>
</I>&gt;<i>     public void recover(IntStream input, RecognitionException re)
</I>&gt;<i>
</I>&gt;<i> Thus, I can't throw an exception in overriding recover method. :-)
</I>
-------------- next part --------------
An HTML attachment was scrubbed...
URL: <A HREF="http://www.antlr.org/pipermail/antlr-interest/attachments/20090531/c3618b54/attachment.html">http://www.antlr.org/pipermail/antlr-interest/attachments/20090531/c3618b54/attachment.html</A> 
</PRE>


<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="034643.html">[antlr-interest] I want to throw an exception and stop parse, please!
</A></li>
	<LI>Next message: <A HREF="034645.html">[antlr-interest] I want to throw an exception and stop parse,	please!
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#34644">[ date ]</a>
              <a href="thread.html#34644">[ thread ]</a>
              <a href="subject.html#34644">[ subject ]</a>
              <a href="author.html#34644">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://www.antlr.org/mailman/listinfo/antlr-interest">More information about the antlr-interest
mailing list</a><br>
</body></html>
